conjurapi/authn_test.go

```diff
@@ -76,6 +76,44 @@ func runRotateAPIKeyAssertions(t *testing.T, tc rotateAPIKeyTestCase, conjur *Cl
 	assert.NoError(t, err)
 }
 
+func TestClient_RotateCurrentUserAPIKey(t *testing.T) {
+	//TODO: This test is ugly. Refactor it into something more concise.
+	t.Run("Rotate the API key of the current user", func(t *testing.T) {
+		// SETUP
+		// Login as admin and rotate alice's API key,
+		// so we can log in as her with her new API key
+		conjur, err := conjurSetup(&Config{}, defaultTestPolicy)
+		assert.NoError(t, err)
+		userApiKey, err := conjur.RotateAPIKey("cucumber:user:alice")
+		assert.NoError(t, err)
+
+		// Login as alice with a mock storage provider to store her API key
+		config := &Config{}
+		config.mergeEnv()
+		conjur, err = NewClientFromKey(*config, authn.LoginPair{Login: "alice", APIKey: string(userApiKey)})
+		assert.NoError(t, err)
+		conjur.storage = &mockStorageProvider{}
+		_, err = conjur.Login("alice", string(userApiKey))
+		assert.NoError(t, err)
+
+		// Ensure we're logged in as alice
```

Comment:
Could probably omit this section since it's essentially testing `NewClientFromKey`. Not much harm in leaving it though if you like, given our recent realizing that we haven't been testing enough!